shells/android/javatests/arcs/storage/ServiceStoreTest.kt

```diff
@@ -0,0 +1,153 @@
+/*
+ * Copyright 2019 Google LLC.
+ *
+ * This code may only be used under the BSD style license found at
+ * http://polymer.github.io/LICENSE.txt
+ *
+ * Code distributed by Google as part of this project is also subject to an additional IP rights
+ * grant found at
+ * http://polymer.github.io/PATENTS.txt
+ */
+
+package arcs.storage
+
+import android.content.ComponentName
+import android.content.ServiceConnection
+import androidx.lifecycle.Lifecycle
+import androidx.test.ext.junit.runners.AndroidJUnit4
+import arcs.crdt.CrdtCount
+import arcs.crdt.parcelables.ParcelableCrdtType
+import arcs.data.CountType
+import arcs.storage.driver.RamDisk
+import arcs.storage.driver.RamDiskDriverProvider
+import arcs.storage.driver.RamDiskStorageKey
+import arcs.storage.parcelables.ParcelableStoreOptions
+import arcs.storage.parcelables.toParcelable
+import arcs.storage.service.BindingContext
+import arcs.storage.service.ConnectionFactory
+import arcs.storage.service.DeferredResult
+import arcs.storage.service.StorageServiceBindingDelegate
+import arcs.storage.service.StorageServiceConnection
+import com.google.common.truth.Truth.assertThat
+import com.nhaarman.mockitokotlin2.mock
+import kotlin.coroutines.coroutineContext
+import kotlinx.coroutines.ExperimentalCoroutinesApi
+import kotlinx.coroutines.runBlocking
+import kotlinx.coroutines.yield
+import org.junit.Before
+import org.junit.Test
+import org.junit.runner.RunWith
+
+/** Tests for the [ServiceStore]. */
+@RunWith(AndroidJUnit4::class)
+@UseExperimental(ExperimentalCoroutinesApi::class)
+class ServiceStoreTest {
+    private lateinit var lifecycle: Lifecycle
+    private val storeOptions = StoreOptions<CrdtCount.Data, CrdtCount.Operation, Int>(
+        RamDiskStorageKey("myData"),
+        ExistenceCriteria.ShouldCreate,
+        CountType()
+    )
+
+    @Before
+    fun setup() {
+        RamDisk.clear()
+        RamDiskDriverProvider()
+        lifecycle = mock()
+    }
+
+    @Test
+    fun getLocalData_getsLocalDataFromService() = runBlocking {
+        val service = buildService(storeOptions.toParcelable(ParcelableCrdtType.Count))
+        val bindingDelegate = buildBindingDelegate(service)
+        val connectionFactory = buildConnectionFactory(bindingDelegate)
+
+        val store = ServiceStore(
+            storeOptions,
+            ParcelableCrdtType.Count,
+            lifecycle,
+            connectionFactory,
+            coroutineContext
+        ).initialize()
```

Comment:
Optional, but I would probably define a helper function for this.

Reply:
for the test cases? that's reasonable... for actual usage, it'll all be handled by the base `Store`'s activate factory, and shouldn't depend on the client to understand how the pieces go together, unless they want to override the defaults.

Reply:
For the test cases.

If you only have the two test cases then it's probably not worth it. I'd made this comment assuming you would add another one.